types/shares.go

```diff
@@ -0,0 +1,148 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
```

Comment:
just a minor note: I would be wary of adding gogoproto further into the repo. The upstream repo will work on removing this dependency in the future.

Reply:
Ah OK, thanks! As soon as upstream will remove this, we'll upgrade too. Here, we only use it to "decorate" (gogo) proto.Messages with a `MarshalDelimited` method. For Tx, and (LL) Message, I have to length prefix manually anyways though.